.github/workflows/tests.yaml

```diff
@@ -0,0 +1,64 @@
+on: [push, pull_request]
```

Comment:
I'm not sure if `pull_request` is necessary here if you already have push. I at least noticed on this PR that we seem to be running 8 checks in total, which is overkill.

Reply:
So I think `pull_request` actually merges in the base branch too, so they're not exactly equivalent. We can probably drop `push` and/or only do it for the `main` branch :shrug:

Reply:
The common pattern is to run tests on pull requests and on pushes to your main / release branches. E.g., Django: https://github.com/django/django/blob/a24fed399ced6be2e9dce4cf28db00c3ee21a21c/.github/workflows/tests.yml#L3-L7

Reply:
Done!

Reply:
...I still see `on: [push, pull_request]`?

Reply:
... my push failed :facepalm: